examples/generative/vae.py

```diff
@@ -90,21 +89,52 @@ def metrics(self):
             self.kl_loss_tracker,
         ]
 
-    def train_step(self, data):
+    def _tf_train_step(self, data):
```

Comment:
This pattern is fine in the general case. In *this* case, I think it's possible to have a much simpler multi-backend implementation by only override `compute_loss`, because the VAE only uses one `self.optimizer.apply` call. `compute_loss` can be written in a way that works with all backends, instead of having 3 backend specific methods.